tools/checklocks/analysis.go

```diff
@@ -495,39 +495,46 @@ func (pc *passContext) checkFunctionCall(call callCommon, fn *types.Func, lff *l
 	// Update all lock state accordingly.
 	pc.postFunctionCallUpdate(call, lff, ls, false /* aliases */)
 
+	// Fast-path handling for sync.(RW)Mutex/Locker helper methods.
+	if fn == nil || fn.Pkg() == nil || len(args) == 0 {
```

Comment:
I'm not sure I understand how this works, but this feels brittle. What is this skipping over?